src/switch_rtp.c

```diff
@@ -1368,15 +1368,21 @@ static void zrtp_event_callback(zrtp_stream_t *stream, unsigned event)
 	switch_rtp_t *rtp_session = zrtp_stream_get_userdata(stream);
 	zrtp_session_info_t zrtp_session_info;
 
-	switch_channel_t *channel = switch_core_session_get_channel(rtp_session->session);
+	switch_channel_t *channel = NULL;
 	switch_event_t *fsevent = NULL;
 	const char *type;
 
+	if (switch_core_session_read_lock(rtp_session->session) != SWITCH_STATUS_SUCCESS) {
```

Comment:
This fails silently the attempt to lock, shouldn't we print some kind of error?

Reply:
There's no harm adding a log in here.